### PR TITLE
[Feature] Support partition live ttl for table (automatic partition part 5)

### LINKED ISSUE #11444: Improve partition management
## Enhancement
Many databases support the function of importing and automatically creating partitions. We can support this function to complete the creation of partitions, and then use ttl to control the deletion of partitions. This ease of use will be better than the previous method.

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #11444

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 15 / 77 (19.48%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|0|5|00.00%|[2002, 2003, 2005, 2006, 2007]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|0|11|00.00%|[249, 250, 252, 253, 254, 255, 256, 257, 259, 260, 263]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|2|00.00%|[1767, 1768]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|3|25|12.00%|[2359, 2360, 2361, 2362, 2363, 3963, 3964, 3965, 3967, 3968, 3969, 3970, 3972, 3973, 3975, 3976, 3977, 3978, 3980, 3981, 3982, 3983]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|7|14.29%|[2403, 2404, 2405, 3121, 3136, 3137]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterTableStatementAnalyzer.java|1|5|20.00%|[160, 161, 162, 163]|
|:large_blue_circle:|com/starrocks/common/util/DynamicPartitionUtil.java|3|11|27.27%|[365, 370, 371, 373, 374, 375, 411, 414]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|1|3|33.33%|[1202, 1203]|
|:large_blue_circle:|com/starrocks/clone/DynamicPartitionScheduler.java|2|3|66.67%|[531]|
|:large_blue_circle:|com/starrocks/catalog/TableProperty.java|4|5|80.00%|[190]|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -3942,6 +3957,31 @@ public void modifyTableDynamicPartition(Database db, OlapTable table, Map<String
...
+        int partitionLiveNumber = -1;
+        if (properties.containsKey(PropertyAnalyzer.PROPERTIES_PARTITION_LIVE_NUMBER)) {
+            try {
+                partitionLiveNumber = PropertyAnalyzer.analyzePartitionLiveNumber(properties, true);
+            } catch (AnalysisException ex) {
+                throw new DdlException(ex.getMessage());
+            }
+        }
+        if (!properties.isEmpty()) {
+            throw new DdlException("Modify failed because unknown properties: " + properties);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:3973]: only modify ttl  without others?

**Review by ABingHuang — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 1 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:3973]: yes

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Pull request merged**